functions/Set-DbaDbFileGroup.ps1

```diff
@@ -0,0 +1,181 @@
+
+    .PARAMETER InputObject
+        Allows piping from Get-DbaDatabase and Get-DbaDbFileGroup.
+
+    .PARAMETER WhatIf
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Data, Database, File, FileGroup, Migration, Partitioning, Table
+        Author: Adam Lancaster https://github.com/lancasteradam
+
+        dbatools PowerShell module (https://dbatools.io)
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Set-DbaDbFileGroup
+
+    .EXAMPLE
+        PS C:\>Set-DbaDbFileGroup -SqlInstance sqldev1 -Database TestDb -FileGroup HRFG1 -Default -AutoGrowAllFiles
+
+        Sets the HRFG1 filegroup to auto grow all files and makes it the default filegroup on the TestDb database on the sqldev1 instance.
+
+    .EXAMPLE
+        PS C:\>Set-DbaDbFileGroup -SqlInstance sqldev1 -Database TestDb -FileGroup HRFG1 -AutoGrowSingleFile
+
+        Sets the HRFG1 filegroup to not auto grow all files on the TestDb database on the sqldev1 instance.
+
+    .EXAMPLE
+        PS C:\>Set-DbaDbFileGroup -SqlInstance sqldev1 -Database TestDb -FileGroup HRFG1 -ReadOnly
+
+        Sets the HRFG1 filegroup to read only on the TestDb database on the sqldev1 instance.
+
+    .EXAMPLE
+        PS C:\>Set-DbaDbFileGroup -SqlInstance sqldev1 -Database TestDb -FileGroup HRFG1 -ReadWrite
+
+        Sets the HRFG1 filegroup to read/write on the TestDb database on the sqldev1 instance.
+
+    .EXAMPLE
+        PS C:\>Get-DbaDatabase -SqlInstance sqldev1 -Database TestDb | Set-DbaDbFileGroup -FileGroup HRFG1 -AutoGrowAllFiles
+
+        Passes in the TestDB database from the sqldev1 instance and sets the HRFG1 filegroup to auto grow all files.
+
+    .EXAMPLE
+        PS C:\>Get-DbaDbFileGroup -SqlInstance sqldev1 -Database TestDb -FileGroup HRFG1 | Set-DbaDbFileGroup -AutoGrowAllFiles
+
+        Passes in the HRFG1 filegroup from the TestDB database on the sqldev1 instance and sets it to auto grow all files.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [string[]]$FileGroup,
+        [switch]$Default,
+        [switch]$ReadOnly,
+        [switch]$ReadWrite,
```

Comment:
I think it might be better to handle these as a single param instead of two switches... this currently doesn't fail and seems to set it to ReadWrite.
`Set-DbaDbFileGroup -SqlInstance mssql1 -Database DatabaseAdmin -FileGroup test2New -ReadWrite -ReadOnly`

You could leave it as a switch `$ReadOnly` and people can use `$ReadOnly:$False` to it back to ReadWrite.. or perhaps make it a string that accepts Enabled or Disabled as values?

Tagging @potatoqualitee and @wsmelton - do you have preferences or do we have a standard we should follow here?